tmpl/setup_env.sh.j2

```diff
@@ -66,32 +65,37 @@ if ${PINNED}; then
         ${CONDA} env update --name ${ENV_NAME} --file requirements/environment.yml || exit
     else
         echo "Dev installation"
-        ${CONDA} env update --name ${ENV_NAME} --file requirements/dev-environment.yml || exit
+        ${CONDA} env update --name ${DEV_ENV_NAME} --file requirements/dev-environment.yml || exit
     fi
 else
+    # First, always create a prod env from unpinned deps
+    # Use the prod env name even if -d has been passed, so the prod env is
+    # exported with its correct name if -e has been passed along with -d
     echo "Unpinned installation"
     ${CONDA} env update --name ${ENV_NAME} --file requirements/requirements.yml || exit
     if ${EXPORT}; then
         echo "Export pinned prod environment"
-        ${CONDA} env export --name ${ENV_NAME} --no-builds > requirements/environment.yml || exit
+        ${CONDA} env export --name ${ENV_NAME} --no-builds | \grep -v '^prefix:' > requirements/environment.yml || exit
     fi
     if ! ${DEV}; then
         echo "WARNING: Unpinned prod installation!!!" >&2
     else
+        # If -d has been passed, install unpinned dev deps on top of prod deps
         echo "Dev installation"
-        ${CONDA} env update --name ${ENV_NAME} --file requirements/dev-requirements.yml || exit
+        ${CONDA} rename --name ${ENV_NAME} ${DEV_ENV_NAME}
+        ${CONDA} env update --name ${DEV_ENV_NAME} --file requirements/dev-requirements.yml || exit
         if ${EXPORT}; then
             echo "Export pinned dev environment"
-            ${CONDA} env export --name ${ENV_NAME} --no-builds > requirements/dev-environment.yml || exit
+            ${CONDA} env export --name ${DEV_ENV_NAME} --no-builds| \grep -v '^prefix:' > requirements/dev-environment.yml || exit
         fi
     fi
 fi
 
-# Install package itself if requested
+# Install package itself
```

Comment:
drop this block? User should do the pip installation manually.

Reply:
That's up to you. Personally, I'd keep it for convenience since the only reason I see not to do it in the script is to avoid it from being included in `conda env export`, but running that manually is not necessary thanks to `-e`. I frequently use an equivalent script to update environments or just create a fresh one, and when it didn't run `pip install -e .` by default, I regularly forgot about it and had to run into an error before remembering. I just don't see a scenario where one would not want to install it... Do you?

But if you feel it's not worth the maintenance, then drop it.